core/runtime/src/main/java/io/quarkus/runtime/CleanableExecutor.java

```diff
@@ -63,8 +63,10 @@ public void run() {
         //threadlocal state. It does not really matter if this does not work, however if these threads are holding
         //state that is stopping things being GC'ed it can help with memory usage
         try {
-            for (int i = 0; i < executor.getMaximumPoolSize(); ++i) {
-                submit(empty);
+            if (!executor.isShutdown()) {
```

Comment:
This is inherently racy (it might shut down after the check but before all the tasks are submitted) but I'd call it "good enough for now" otherwise.